FEXCore/Source/Utils/CPUInfo.cpp

```diff
@@ -14,17 +15,31 @@
 namespace FEXCore::CPUInfo {
 #ifndef _WIN32
 uint32_t CalculateNumberOfCPUs() {
-  char Tmp[PATH_MAX];
+  static uint32_t NumCPUs {};
+  if (NumCPUs) {
+    return NumCPUs;
+  }
+
+#define CPUSTRING "/sys/devices/system/cpu/cpu"
+  constexpr size_t CPUStringLength = std::char_traits<char>::length(CPUSTRING);
+
+  // Support up to 8 digits of CPU cores plus the null terminator.
+  char Tmp[CPUStringLength + 1 + 8] = CPUSTRING;
+#undef CPUSTRING
+  static_assert(CPUStringLength == 27, "String length changed. Check that this is correct.");
   size_t CPUs = 1;
 
   for (;; ++CPUs) {
-    auto Size = fmt::format_to_n(Tmp, sizeof(Tmp), "/sys/devices/system/cpu/cpu{}", CPUs);
-    Tmp[Size.size] = 0;
+    auto Res = std::to_chars(&Tmp[CPUStringLength], &Tmp[sizeof(Tmp)], CPUs);
+    // null terminate the string.
+    Res.ptr[0] = '\0';
+
     if (!FHU::Filesystem::Exists(Tmp)) {
       break;
     }
   }
 
+  NumCPUs = CPUs;
   return CPUs;
```

Comment:
To keep control flow straightforward, this can just be moved inside of the failed `if (Exists)` check above.

Reply:
Moved